lib/routes/user/marketplace/vendor_webview.dart

```diff
@@ -62,53 +64,85 @@ class VendorWebViewPageState extends State<VendorWebViewPage> {
         // Wait for memory image to load
         Timer(Duration(milliseconds: 200), () {
           // Hide Webview to interact with payment request dialog
-          _widgetWebview.hide(); 
+          _widgetWebview.hide();
           beforeCompleter.complete();
         });
       });
-    }); 
+    });
 
-    return beforeCompleter.future; 
+    return beforeCompleter.future;
   }
 
   @override
   void didChangeDependencies() {
     if (!_isInit) {
       var invoiceBloc = AppBlocsProvider.of<InvoiceBloc>(context);
       var accountBloc = AppBlocsProvider.of<AccountBloc>(context);
-      _weblnHandlers = WeblnHandlers(context, accountBloc, invoiceBloc, onBeforeCallHandler);
+      if (widget.listenInvoices) {
+        _weblnHandlers = WeblnHandlers(context, accountBloc, invoiceBloc, onBeforeCallHandler);
 
-      String loadedURL;
-      _widgetWebview.onStateChanged.listen((state) async {
-        if (state.type == WebViewState.finishLoad && loadedURL != state.url) {          
-          loadedURL = state.url;
-          _widgetWebview.evalJavascript(await _weblnHandlers.initWeblnScript);
-        }
-      });
-      
-      _postMessageListener = _widgetWebview.onPostMessage.listen((msg) {        
-        if (msg != null ) {         
-          var postMessage = (widget._title == "ln.pizza") ? {"action": "sendPayment", "payReq": msg} : JSON.jsonDecode(msg);                    
-          _weblnHandlers.handleMessage(postMessage)
-            .then((resScript){
+        String loadedURL;
+        _widgetWebview.onStateChanged.listen((state) async {
+          if (state.type == WebViewState.finishLoad && loadedURL != state.url) {
+            loadedURL = state.url;
+            _widgetWebview.evalJavascript(await _weblnHandlers.initWeblnScript);
+          }
+        });
+
+        _postMessageListener = _widgetWebview.onPostMessage.listen((msg) {
+          if (msg != null) {
+            var postMessage = (widget._title == "ln.pizza") ? {"action": "sendPayment", "payReq": msg} : JSON.jsonDecode(msg);
+            _weblnHandlers.handleMessage(postMessage).then((resScript) {
               if (resScript != null) {
                 _widgetWebview.evalJavascript(resScript);
                 _widgetWebview.show();
                 setState(() {
                   _screenshotData = null;
                 });
               }
-            });            
-        }
-      });
+            });
+          }
+        });
+      } else {
+        String loadedURL;
```

Comment:
I don't think the internals of moonpay logic belongs here as they are not related to VendorWebView.
I also see that most of the logic in this class is not relevant to moonpay as well so I think it is better to create a separate widget "MoonpayWebview" with the specific logic.

Reply:
[Fixed](https://github.com/breez/breezmobile/pull/162/commits/ec80a3164e05ea547c90c5d39dd519e0563b431e)